xtractmime/__init__.py

```diff
@@ -67,35 +147,37 @@ def extract_mime(
     http_origin: bool = True,
     no_sniff: bool = False,
     extra_types: Optional[Tuple[Tuple[bytes, bytes, Set[bytes], str], ...]] = None,
+    supported_types: Set[str] = None,
 ) -> str:
     extra_types = extra_types or tuple()
     supplied_type = content_types[-1] if content_types else None
     check_for_apache = http_origin and supplied_type in _APACHE_TYPES
-    resource_header = memoryview(body) if len(body) < 1445 else memoryview(body)[:1445]
+    resource_header = memoryview(body)[:1445]
 
     if supplied_type in (None, "unknown/unknown", "application/unknown", "*/*"):
-        _find_unknown_mimetype(body, not no_sniff)
+        _find_unknown_mimetype(resource_header, not no_sniff, extra_types)
 
     if no_sniff:
         return supplied_type
 
     if check_for_apache:
-        return _sniff_mislabled_binary(body)
+        return _sniff_mislabled_binary(resource_header)
 
-    if supplied_type[-4:] is "+xml" or supplied_type in ("text/xml", "application/xml"):
+    if supplied_type.endswith("+xml") or supplied_type in {"text/xml", "application/xml"}:
         return supplied_type
 
-    if supplied_type is "text/html":
-        return _compare_feed_html(body)
-
-    matched_type = is_image(resource_header)
-
-    if matched_type:
-        return matched_type
+    if supplied_type.startswith("text/html"):
+        return _sniff_mislabled_feed(resource_header)
 
-    matched_type = is_audio_video(resource_header)
+    if supplied_type.startswith("image/"):
+        matched_type = get_image_mime(resource_header)
+        if matched_type in supported_types:
+            return matched_type
 
-    if matched_type:
-        return matched_type
+    video_types = ("audio/","video/")
+    if supplied_type.startswith(video_types) or supplied_type.startswith("application/ogg"):
```

Comment:
I don’t think `startswith` supports multiple values, I believe you need to use it twice instead. And only use it to check the beginning of a string, use `supplied_type == "application/ogg"` instead.

Reply:
I think its working
```
>>> a = "audio/"
>>> b = ("audio/","video/")
>>> a.startswith(b)
True
>>> a = "video/"
>>> a.startswith(b)
True
>>> a = "text/"
>>> a.startswith(b)
False
>>> 


```

Reply:
Today I learn, good to know!